src/__tests__/role-helpers.js

```diff
@@ -19,7 +19,7 @@ function setup() {
 <header data-testid="a-header">Banner header</header>
 <section aria-label="a region" data-testid='named-section'>
   <a href="http://whatever.com" data-testid="a-link">link</a>
-  <a>invalid link</a>
+  <a data-testid="invalid-link">invalid link</a>
```

Comment:
Maybe we should add a test for `img` role with `alt` value (role img) and with an empty `alt` (role presentation) to see that we get the corresponding role for each. I know that this is not logic that we own but I believe it will help us to understand breakage later on.
Wdyt?

Reply:
Agreed, good to have some coverage on this.

Interestingly adding the test cases highlighted an issue - a combination of the empty value issue (see also https://github.com/testing-library/dom-testing-library/pull/1236) as well as a limitation in the "specificity" sorting algorithm. I've pushed amends for both.

Reply:
I'm not sure if this was an issue before or that's a new `aria-query` feature and if so then we can understand why it won't work to begin with.

Reply:
I've just trialled the empty `alt=""` test case on `main` and it currently reports `img` for the role so I believe it's an existing issue with this lib rather than a consequence of the `aria-query` bump (as the old `5.1.3` version of `aria-query` had this ruleset, `5.2.0` regressed it).